src/ui/command.rs

```diff
@@ -286,12 +286,19 @@ fn parse_key_event(
     buffer: &mut CommandBuffer,
     writer: &Sender<Event>,
     tts_ctrl: &mut Arc<Mutex<TTSController>>,
+    save_history: bool,
```

Comment:
save_history is Arc<AtomicBool> in Session object, so we can safely share it among threads. I think in future we can refactor everything to share full Settings object as sort of context for many threads and components.